core/templates/dev/head/components/StaticGraphDirective.js

```diff
@@ -0,0 +1,329 @@
+                  'M' + startx + ' ' + starty + ' Q ' + midx + ' ' + midy +
+                  ' ' + endx + ' ' + endy);
+
+                // Style links if link properties and style mappings are
+                // provided
+                if (links[i].hasOwnProperty('linkProperty') &&
+                    $scope.linkPropertyMapping) {
+                  if ($scope.linkPropertyMapping.hasOwnProperty(
+                        links[i].linkProperty)) {
+                    $scope.augmentedLinks[i].style = (
+                      $scope.linkPropertyMapping[links[i].linkProperty]);
+                  }
+                }
+              }
+            }
+
+            var nodeStrokeWidth = '1';
+
+            var getNodeFillOpacity = function(nodeId) {
+              return $scope.opacityMap ? $scope.opacityMap[nodeId] : 0.5;
+            };
+
+            $scope.getNodeTitle = function(node) {
+              var tooltip = node.label;
+              if (node.hasOwnProperty('secondaryLabel')) {
+                tooltip += ' ' + node.secondaryLabel;
+              }
+              return tooltip;
+            };
+
+            $scope.canNavigateToNode = function(nodeId) {
+              //return nodeId !== $scope.currentStateId();
+              return true;
+            };
+
+            $scope.getTruncatedLabel = function(nodeLabel) {
+              return $filter('truncate')(nodeLabel, MAX_NODE_LABEL_LENGTH);
+            };
+
+            // Update the nodes.
+            $scope.nodeList = [];
+            for (var nodeId in nodeData) {
+              nodeData[nodeId].style = (
+                'stroke-width: ' + nodeStrokeWidth + '; ' +
+                'fill-opacity: ' + getNodeFillOpacity(nodeId) + ';');
+
+              if ($scope.nodeFill) {
+                nodeData[nodeId].style += ('fill: ' + $scope.nodeFill + '; ');
+              }
+
+              // Color nodes
+              if ($scope.nodeColors) {
+                nodeData[nodeId].style += (
+                  'fill: ' + $scope.nodeColors[nodeId] + '; ');
+              }
+
+              // Add secondary label if it exists
+              if ($scope.nodeSecondaryLabels) {
+                if ($scope.nodeSecondaryLabels.hasOwnProperty(nodeId)) {
+                  nodeData[nodeId].secondaryLabel = (
+                    $scope.nodeSecondaryLabels[nodeId]);
+                  nodeData[nodeId].height *= 1.1;
+                }
+              }
+
+              var currentNodeIsTerminal = (
+                $scope.finalStateIds.indexOf(nodeId) !== -1);
+
+              nodeData[nodeId].nodeClass = (
+                // currentNodeIsTerminal ? 'terminal-node' :
+                // nodeId === $scope.currentStateId() ? 'current-node' :
+                // nodeId === initStateId ? 'init-node' :
+                // !(nodeData[nodeId].reachable &&
+                //  nodeData[nodeId].reachableFromEnd) ? 'bad-node' :
+                                                       'normal-node');
```

Comment:
should I remove these the commented thing. Can these be useful for a static graph scenario? I manually created two graph each in statistics tab and history tab using both StateGraphVisualizationDirective and StaticGraphDirective and didn't find any difference.